mods/pageviews.js

```diff
@@ -0,0 +1,313 @@
+'use strict';
+
+/**
+ * Pageviews API module
+ *
+ * This API serves pre-aggregated pageview statistics from Cassandra
+ */
+
+var URI = require('swagger-router').URI;
+
+var fs = require('fs');
+var yaml = require('js-yaml');
+var path = require('path');
+var spec = yaml.safeLoad(fs.readFileSync(path.join(__dirname, '/pageviews.yaml')));
+var rbUtil = require('../lib/rbUtil');
+
+
+// Pageviews Service
+function PJVS(options) {
+    this.options = options;
+    this.log = options.log || function() {};
+}
+
+
+var tables = {
+    article: 'pageviews.per.article',
+    project: 'pageviews.per.project',
+    tops: 'top.pageviews',
+};
+var tableURI = function(domain, tableName) {
+    return new URI([domain, 'sys', 'table', tableName, '']);
+};
+var tableSchemas = {
+    article: {
+        table: tables.article,
+        version: 1,
+        attributes: {
+            project: 'string',
+            article: 'string',
+            access: 'string',
+            agent: 'string',
+            granularity: 'string',
+            // the hourly timestamp will be stored as YYYYMMDDHH
+            timestamp: 'string',
+            views: 'int'
+        },
+        index: [
+            { attribute: 'project', type: 'hash' },
+            { attribute: 'article', type: 'hash' },
+            { attribute: 'access', type: 'hash' },
+            { attribute: 'agent', type: 'hash' },
+            { attribute: 'granularity', type: 'hash' },
+            { attribute: 'timestamp', type: 'range', order: 'asc' },
+        ]
+    },
+    project: {
+        table: tables.project,
+        version: 1,
+        attributes: {
+            project: 'string',
+            access: 'string',
+            agent: 'string',
+            granularity: 'string',
+            // the hourly timestamp will be stored as YYYYMMDDHH
+            timestamp: 'string',
+            views: 'int'
+        },
+        index: [
+            { attribute: 'project', type: 'hash' },
+            { attribute: 'access', type: 'hash' },
+            { attribute: 'agent', type: 'hash' },
+            { attribute: 'granularity', type: 'hash' },
+            { attribute: 'timestamp', type: 'range', order: 'asc' },
+        ]
+    },
+    tops: {
+        table: tables.tops,
+        version: 1,
+        attributes: {
+            project: 'string',
+            access: 'string',
+            year: 'string',
+            month: 'string',
+            day: 'string',
+            // format for this is a json array: [{rank: 1, article: <<title>>, views: 123}, ...]
+            articles: 'string'
+        },
+        index: [
+            { attribute: 'project', type: 'hash' },
+            { attribute: 'access', type: 'hash' },
+            { attribute: 'year', type: 'hash' },
+            { attribute: 'month', type: 'hash' },
+            { attribute: 'day', type: 'hash' },
+        ]
+    }
+};
+
+/**
+ * general handler functions */
+var queryResponser = function(res) {
```

Comment:
Nit: should probably be `queryResponder` (sorry, my OCD doesn't allow me to ignore this) :P